scripts/trim_indel_ends.py

```diff
@@ -0,0 +1,402 @@
+parser.add_argument('-m', '--'+optList[0], type=str, metavar='', default='lenient', help='string\tMode of trimming. "Strict" only trims when sequences match exactly after trimming; "Lenient" trims whenever prefix and suffix indel numbers below threshold, defauls to "Lenient"')
+parser.add_argument('-t', '--'+optList[1], type=int, metavar='', default=10000, help='int\tThreshold for trimming, defaults to 10000 (trim every prefix and suffix)')
+
+# strict positional/optional arguments checking
+argsDict = vars(parser.parse_args())
+sys.stdout.write('\n' + datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - ' + sys.argv[0] + ' - Parsing Input Arguements...\n\n'); sys.stdout.flush()
+for key, value in argsDict.items():
+    if key in posList: sys.stdout.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - Required Argument - ' + key +': '+ str(value) + '\n'); sys.stdout.flush()
+    if key in optList: sys.stdout.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - Optional Argument - ' + key +': '+ str(value) + '\n'); sys.stdout.flush()
+    vars()[key] = value # assign values of arguments into shorthand global variables
+sys.stdout.write('\n'); sys.stdout.flush()
+
+def extract_edits(cigar_str) -> list:
+    """Given a cigar string, return separated edits with counts
+
+    Args:
+        cigar_str (str):
+
+    Returns:
+        list: a list of tuples of (count, edit_type)
+    """
+    all_edits = [(int(s[:-1]), s[-1]) for s in re.findall(r'\d+[=,I,D,X]', cigar_str)]
+    return all_edits
+
+def ed_from_cigar(cigar_str) -> int:
+    """Compute edit distance from cigar strings
+
+    Args:
+        cigar_str (str): cigar string
+
+    Returns:
+        int: edit distance that is equal to total length of the string minus the nu
+    """
+    all_edits = extract_edits(cigar_str)
+    total_length = sum([e[0] for e in all_edits])
+    match_count = sum([e[0] for e in all_edits if e[1] == "="])
+    return total_length - match_count
+
+def get_len_type(indel_list:list) -> tuple[str, str]:
+    """Given a list of indels, return the total length and type
+
+    Args:
+        indel_list (list): list of indels
+
+    Returns:
+        tuple[str, str]: type, length
+    """
+
+    indel_type = ""
+    indel_length = 0
+    if len(indel_list) != 0:
+        for edit in indel_list:
+            if indel_type != "":
+                assert indel_type == edit[-1]
+            else:
+                indel_type = edit[-1]
+            indel_length += edit[0]
+    return indel_type, indel_length
+
+def get_prefix_and_suffix_indels(cigar_str:str, mode="lenient") -> tuple:
+    """Get indel edits from the prefix and suffix of the cigar string
+
+    Args:
+        cigar_str (str): cigar string
+
+    Returns:
+        tuple: (list, list): prefix list of indels and suffix list of indels
+    """
+    all_edits = extract_edits(cigar_str)
+    prefix_indels = []
+    suffix_indels = []
+    i = 0
+    for i in range(len(all_edits)):
```

Comment:
I'm worried about the `break` logic here. 

@ame9yu Could you do two things here:

(1) I think I follow the logic, but it's good to summarize in code comments. Could you add 1-2 comments in the code before the for loops explaining what you're doing here?

(2) For future unit tests, please output an expected example....perhaps in the python docs-comments above

Reply:
Example added and comments added!